src/main/scala/where/Condition.scala

```diff
@@ -36,13 +36,21 @@ case class NullCondition(reference: Reference, check: NullCheck) extends Conditi
     reference.getReferenceable.toSet
 }
 
-case class PredicateCondition(predicate: Predicate, projection: Collection, where: AnyNode => Condition) extends Condition {
+case class PredicateCondition(
+  predicate: Predicate,
+  projection: Collection,
+  where: ObjectReference => Where
+) extends Condition {
   def toQuery(referenceableMap: ReferenceableMap): String = {
     val identifier = "x"
-    val node = AnyNode()
-    val fakeMap: ReferenceableMap = Map(node -> identifier)
-    val conditionString = where(node).toQuery(fakeMap)
-    s"${predicate.toQuery} ($identifier in ${projection.toQuery(referenceableMap)} where $conditionString)"
+    val referenceable = AnyNode()
+    val adjustedMap = referenceableMap + (referenceable -> identifier)
```

Comment:
i like acknowledging `ReferenceableMap` is its own structure. good path to robustifying too, in the future